webapp/client/src/pages/LoginPage.js

```diff
@@ -1,60 +1,94 @@
 import PropTypes from "prop-types";
 import React from "react";
+import { useTranslation } from "react-i18next";
+import FormFieldIdNr from "../components/FormFieldIdNr";
+import FormFieldUnlockCode from "../components/FormFieldUnlockCode";
 import FormHeader from "../components/FormHeader";
+import FormRowCentered from "../components/FormRowCentered";
 import StepForm from "../components/StepForm";
 import StepHeaderButtons from "../components/StepHeaderButtons";
 
-export default function LoginPage({
-  prevUrl,
-  backLinkText,
-  stepTitle,
-  stepIntro,
-  submitUrl,
-  overviewButton,
-  csrfToken,
-  explanatoryButtonText,
-  nextButtonLabel,
-}) {
+export default function LoginPage({ backLink, stepHeader, form, fields }) {
+  const { t } = useTranslation();
+
   return (
-    <React.Fragment>
-      <StepHeaderButtons backLinkUrl={prevUrl} backLinkText={backLinkText} />
-      <FormHeader title={stepTitle} intro={stepIntro} />
-      <StepForm
-        {...{
-          submitUrl,
-          overviewButton,
-          csrfToken,
-          explanatoryButtonText,
-          nextButtonLabel,
-        }}
-      />
-    </React.Fragment>
+    <>
+      <StepHeaderButtons {...backLink} />
+      <FormHeader {...stepHeader} />
+      <StepForm {...form}>
+        <FormRowCentered>
+          <FormFieldIdNr
+            autofocus
+            required
+            fieldName="idnr"
+            // TODO: is the fieldId ever different from the fieldName?
+            fieldId="idnr"
+            values={fields.idnr.value}
+            label={{
+              text: t("unlockCodeActivation.idnr.labelText"),
+            }}
+            details={{
+              title: t("unlockCodeActivation.idnr.help.title"),
+              text: t("unlockCodeActivation.idnr.help.text"),
+            }}
+            errors={fields.idnr.errors}
+          />
+        </FormRowCentered>
+        <FormRowCentered>
+          <FormFieldUnlockCode
+            required
+            fieldName="unlock_code"
+            fieldId="unlock_code" // field.id
+            values={fields.unlockCode.value}
+            label={{
+              text: t("unlockCodeActivation.unlockCode.labelText"),
+            }}
+            details={{
+              title: t("unlockCodeActivation.unlockCode.help.title"),
+              text: t("unlockCodeActivation.unlockCode.help.text"),
+            }}
+            errors={fields.unlockCode.errors}
+          />
+        </FormRowCentered>
+      </StepForm>
+    </>
   );
 }
 
+const fieldPropType = PropTypes.exact({
+  // field._value()
+  value: PropTypes.any,
+  // field.errors
+  errors: PropTypes.arrayOf(PropTypes.string),
+});
```

Comment:
What are these comments for?

Reply:
Reminders for me what the values for these fields need to be.